errors.go

```diff
@@ -0,0 +1,56 @@
+package filclient
+
+import "fmt"
+
+type ErrorCode int
+
+const (
+	// Failed to connect to a miner.
+	ErrMinerConnectionFailed ErrorCode = iota
```

Comment:
I would include an 'Unknown' one as the first value, so that if its unset by accident, we dont think its a connection failure.